lib/V8/v8-utils.cpp

```diff
@@ -202,6 +202,85 @@ static void CreateErrorObject(v8::Isolate* isolate, int errorNumber,
   }
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// @brief set a point in time after which we will abort external connection
+////////////////////////////////////////////////////////////////////////////////
+static std::atomic<double> executionDeadline(0.0);
+void setExecutionDeadlineInMS(uint64_t timeout) {
+  if (timeout == 0) {
+    executionDeadline = 0.0;
+  } else {
+    executionDeadline = TRI_microtime() + timeout / 1000;
+  }
+}
+
+static void JS_SetExecutionDeatlineTo(v8::FunctionCallbackInfo<v8::Value> const& args) {
+  TRI_V8_TRY_CATCH_BEGIN(isolate);
+  v8::HandleScope scope(isolate);
+
+  // extract arguments
+  if (args.Length() != 1) {
+    TRI_V8_THROW_EXCEPTION_USAGE("SetGlobalExecutionDeadlineTo(<timeout>)");
+  }
+
+  auto when = executionDeadline.load();
+  auto now = TRI_microtime();
+
+  auto n = TRI_ObjectToUInt64(isolate, args[0], false);
+  setExecutionDeadlineInMS(n);
+
+  TRI_V8_RETURN_BOOL((when > 0.00001) && (now - when > 0.0) );
+  TRI_V8_TRY_CATCH_END
+}
+
+bool isExecutionDeadlineReached(v8::FunctionCallbackInfo<v8::Value> const& args) {
+  auto when = executionDeadline.load();
+  if (when < 0.00001) {
+    return false;
+  }
+  auto now = TRI_microtime();
+  if (now < when) {
+    return false;
+  }
+
+  auto isolate = args.GetIsolate();
+  TRI_CreateErrorObject(isolate, TRI_ERROR_DISABLED, "Deadline for remote reached!", true);
+  return true;
+}
+
+double correctTimeoutToExecutionDeadlineS(double timeoutSeconds) {
```

Comment:
Does this function do the same as the function below? If so is there no way to express one with help of the other?

Reply:
this version can probably be erased along with the simple http client - thus I don't want to spend more time on this.